src/main/java/chatlinitation/ChatViewmodel.java

```diff
@@ -0,0 +1,44 @@
+package chatlinitation;
+
+import entities.PrivateChat;
+
+/**
+ * chatviewmodel  connects our ChatView UI to our Entities.
+ * It provide data to the view, so that view can put that data on the screen.
+ * containing the message content and associated metadata
+ * this class contain privatechat, recipeint user and Textmessge.
+ */
+
+public class ChatViewmodel {
+
+    PrivateChat privatechat;
+    User recipient;
+    TextMessage txtmessage;
+
+
+ // constructor of the viewmodel
+    public ChatViewmodel() {
+        this.privatechat = privatechat;
+        this.recipient=recipient;
+        this.txtmessage =  txtmessage;
+    }
+
+
+
+    // when we click the  addbutton for username.our private chat should update it's Recipientusername.
+    public void setRecipientUsername(String recipientUsername) {
+        this.privatechat.setRecipientUsername(recipientUsername) ;
+    }
+
+
+    //when  type an input click send button , the input should set the txtmessge content
+    //and privatechat should  the txtmessage to conversation.
+    public void setMessage(String content) {
+        this.txtmessage.setMsgContent(content) ;
+        privatechat.addtoconvHist(txtmessage);
```

Comment:
I mentioned this to you verbally already (and you have already made the changes), but here is the comment as a written record. Your previous ChatViewModel is more of an interactor. Since it's an interactor, it can directly manipulate entities without problems.

Reply:
Yes Correct I create two Chat_interactors instead. This is a better clean architecture. Thank James